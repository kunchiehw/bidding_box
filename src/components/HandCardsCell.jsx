import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Segment, Divider, Button } from 'semantic-ui-react';
import getMessage from '../i18n/messages';
import { EW_SEATS, PARTICIPANTS } from '../utils/constants';
import { suitColor } from '../utils/helpers';


const propTypes = {
  seat: PropTypes.oneOf(EW_SEATS).isRequired,
  playerRole: PropTypes.oneOf(PARTICIPANTS).isRequired,
  playerId: PropTypes.string,
  playerHand: PropTypes.shape({
    SPADES: PropTypes.string.isRequired,
    HEARTS: PropTypes.string.isRequired,
    DIAMONDS: PropTypes.string.isRequired,
    CLUBS: PropTypes.string.isRequired,
  }),
  bidSeqIsEnded: PropTypes.bool.isRequired,
  handleClickToSit: PropTypes.func.isRequired,
};

const defaultProps = {
  playerHand: null,
  playerId: null,
};

class HandCardsCell extends PureComponent {
  render() {
    const idCell = (this.props.playerId) ? (
      <div className="id-cell">
        {this.props.playerId}
      </div>
    ) : (
      <Button onClick={() => this.props.handleClickToSit(this.props.seat)}>
        {`Click to sit ${this.props.seat}`}
      </Button>
    );

    const cardCell = ((this.props.playerHand) &&
    (this.props.bidSeqIsEnded || this.props.playerRole === 'OBSERVER' || this.props.playerRole === this.props.seat)) ? (
      <div className="card-cell">
        <div className="card-cell-content">
          <div>
            <span style={{ color: suitColor('SPADES') }}> {getMessage('SPADES')} </span>
            <span> {this.props.playerHand.SPADES} </span>
          </div>
          <div>
            <span style={{ color: suitColor('HEARTS') }}> {getMessage('HEARTS')} </span>
            <span> {this.props.playerHand.HEARTS} </span>
          </div>
          <div>
            <span style={{ color: suitColor('DIAMONDS') }}> {getMessage('DIAMONDS')} </span>
            <span> {this.props.playerHand.DIAMONDS} </span>
          </div>
          <div>
            <span style={{ color: suitColor('CLUBS') }}> {getMessage('CLUBS')} </span>
            <span> {this.props.playerHand.CLUBS} </span>
          </div>
        </div>
      </div>
      ) : (
        <div className="card-cell" />
      );

    return (
      <Segment className="hand-cards-display-cell">
        {idCell}
        <Divider />
        {cardCell}
      </Segment>
    );
  }
}

HandCardsCell.propTypes = propTypes;
HandCardsCell.defaultProps = defaultProps;

export default HandCardsCell;
