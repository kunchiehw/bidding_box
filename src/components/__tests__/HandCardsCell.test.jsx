import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HandCardsCell from '../HandCardsCell';
import { EW_SEATS, PARTICIPANTS, BOOLEANS } from '../../utils/constants';

describe('HandCardsCell render', () => {
  const ids = ['', 'testId'];
  const hands = [
    null,
    {
      SPADES: 'AKQ', HEARTS: 'AKQJT98', DIAMONDS: '432', CLUBS: '',
    },
  ];

  EW_SEATS.forEach((seat) => {
    PARTICIPANTS.forEach((playerRole) => {
      ids.forEach((playerId) => {
        hands.forEach((playerHand) => {
          BOOLEANS.forEach((bidSeqIsEnded) => {
            it(`seat: ${seat}, playerRole: ${playerRole}, playerId: ${playerId},
              playerHand: ${playerHand}, bidSeqIsEnded: ${bidSeqIsEnded}`, () => {
              const handCardsCell = shallow(<HandCardsCell
                seat={seat}
                playerRole={playerRole}
                playerId={playerId}
                playerHand={playerHand}
                bidSeqIsEnded={bidSeqIsEnded}
                handleClickToSit={() => {}}
              />);
              expect(toJson(handCardsCell)).toMatchSnapshot();
            });
          });
        });
      });
    });
  });
});
