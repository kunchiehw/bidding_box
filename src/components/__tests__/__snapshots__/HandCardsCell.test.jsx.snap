// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: ,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: ,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: testId,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: EAST, playerId: testId,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: ,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: ,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: testId,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WATCHER, playerId: testId,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: ,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: ,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit EAST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: testId,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: EAST, playerRole: WEST, playerId: testId,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: ,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: ,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: testId,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: EAST, playerId: testId,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: ,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: ,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: testId,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WATCHER, playerId: testId,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: ,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: ,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: ,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <Button
    as="button"
    onClick={[Function]}
  >
    Click to sit WEST
  </Button>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: testId,
              playerHand: [object Object], bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  >
    <div
      className="card-cell-content"
    >
      <div>
        <span
          style={
            Object {
              "color": "blue",
            }
          }
        >
           
          ♠
           
        </span>
        <span>
           
          AKQ
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
           
          ♥
           
        </span>
        <span>
           
          AKQJT98
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "orange",
            }
          }
        >
           
          ♦
           
        </span>
        <span>
           
          432
           
        </span>
      </div>
      <div>
        <span
          style={
            Object {
              "color": "green",
            }
          }
        >
           
          ♣
           
        </span>
        <span>
           
           
        </span>
      </div>
    </div>
  </div>
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: testId,
              playerHand: null, bidSeqIsEnded: false 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;

exports[`HandCardsCell render seat: WEST, playerRole: WEST, playerId: testId,
              playerHand: null, bidSeqIsEnded: true 1`] = `
<Segment
  className="hand-cards-display-cell"
>
  <div
    className="id-cell"
  >
    testId
  </div>
  <Divider />
  <div
    className="card-cell"
  />
</Segment>
`;
